packages/tenant-process/src/services/tenantService.ts

```diff
@@ -383,6 +387,70 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async addDeclaredAttribute({
+      tenantAttributeSeed,
+      authData,
+      correlationId,
+    }: {
+      tenantAttributeSeed: ApiDeclaredTenantAttributeSeed;
+      authData: AuthData;
+      correlationId: string;
+    }): Promise<Tenant> {
+      logger.info(
+        `Add declared attribute ${tenantAttributeSeed.id} to requester tenant ${authData.organizationId}`
+      );
+      const targetTenant = await retrieveTenant(
+        authData.organizationId,
+        readModelService
+      );
+
+      const maybeDeclaredTenantAttribute = targetTenant.data.attributes.find(
+        (attr) => attr.id === tenantAttributeSeed.id
+      );
+
+      // eslint-disable-next-line functional/no-let
+      let updatedTenant: Tenant = {
+        ...targetTenant.data,
+        updatedAt: new Date(),
+      };
+      if (!maybeDeclaredTenantAttribute) {
+        // assigning attribute for the first time
+        updatedTenant = {
+          ...updatedTenant,
+          attributes: [
+            ...targetTenant.data.attributes,
+            {
+              id: unsafeBrandId(tenantAttributeSeed.id),
+              type: tenantAttributeType.DECLARED,
+              assignmentTimestamp: new Date(),
+              revocationTimestamp: undefined,
+            },
```

Comment:
Shouldn't here be some logic that checks if the declared attribute actually exists in the read model? I checked in the Scala repo and it's also not there. Should we add it?

We are trusting the user input too much here.

cc @AsterITA  @taglioni-r @galales

Reply:
Sure, makes sense!